### PR TITLE
seed timeout randomness from crypto/rand

### PR DESCRIPTION
**Pull request by superfell**
to ensure different nodes use different sequences of random timeouts. Fixes issue #50

### CODE REVIEW

File: util.go [hunk 1]
```diff
@@ -5,18 +5,29 @@ import (
...
 	"math/rand"
 	"time"
 
 	"github.com/hashicorp/go-msgpack/codec"
 )
 
+var rnd = rand.New(rand.NewSource(newSeed()))
+
+func newSeed() int64 {
+	r, err := crand.Int(crand.Reader, big.NewInt(2^63))
```
Comment (armon) [inline util.go]: Instead of `2^63` lets use `math.MaxInt64`

File: util.go [hunk 2]
```diff
@@ -5,18 +5,29 @@ import (
...
+	"math/big"
 	"math/rand"
 	"time"
 
 	"github.com/hashicorp/go-msgpack/codec"
 )
 
+var rnd = rand.New(rand.NewSource(newSeed()))
+
+func newSeed() int64 {
```
Comment (armon) [inline util.go]: Can you add a doc string and a test to verify that the first 1000 results are unique.

**Comment by superfell** [comment]
Are you ok with this in multiple commits, or do you want me to squash it down to one ?

**Comment by armon** [comment]
Thanks! This is great!

**Comment by armon** [comment]
We may be able to use this in other places to use `rnd` instead of `rand`

**Pull request merged**